tests/plots/test_period_on_period.py

```
@@ -0,0 +1,157 @@
+
+
+@pytest.mark.usefixtures("_mock_color_generators", "_mock_gu_functions")
+def test_overlapping_periods_with_source_text(sample_dataframe):
+    """Test overlapping periods with source text added."""
+    periods = [("2023-01-01", "2023-01-05")]
+    source_text = "Source: Sales Data"
+
+    ax = plot(
+        df=sample_dataframe,
+        x_col="date",
+        value_col="value",
+        periods=periods,
+        source_text=source_text,
+    )
+
+    gu.add_source_text.assert_called_once_with(ax=ax, source_text=source_text)
+    assert isinstance(ax, Axes)
+
+
+@pytest.mark.usefixtures("_mock_color_generators", "_mock_gu_functions")
+def test_overlapping_periods_with_legend_title_and_outside(sample_dataframe):
+    """Test overlapping periods with legend title and moved legend."""
+    periods = [("2023-01-01", "2023-01-05"), ("2023-01-06", "2023-01-10")]
+    legend_title = "Periods"
+
+    ax = plot(
+        df=sample_dataframe,
+        x_col="date",
+        value_col="value",
+        periods=periods,
+        move_legend_outside=True,
+        legend_title=legend_title,
+    )
+
+    gu.standard_graph_styles.assert_any_call(
+        ax=ax,
+        title=None,
+        x_label="date",
+        y_label="value",
+        legend_title=legend_title,
+        move_legend_outside=True,
+    )
+    assert isinstance(ax, Axes)
+
+
+@pytest.mark.usefixtures("_mock_color_generators", "_mock_gu_functions")
+def test_overlapping_periods_raises_on_empty_periods(sample_dataframe):
+    """Test overlapping periods with a ValueError is raised if an empty list of periods is passed."""
+    with pytest.raises(ValueError, match="The 'periods' list must contain at least one"):
+        plot(
+            df=sample_dataframe,
+            x_col="date",
+            value_col="value",
+            periods=[],
+        )
+
+
+@pytest.mark.usefixtures("_mock_color_generators", "_mock_gu_functions")
+def test_overlapping_periods_skips_empty_period_df(sample_dataframe):
```

Do I understand it correctly, that if only a single period is given, then no lines will be added to the plot? If that's the case I'd say it should return an error as normal behaviour.